cmds/dump.c

```diff
@@ -1209,6 +1215,130 @@ static void print_flame_footer(struct uftrace_dump_ops *ops,
 	graph_remove_task();
 }
 
+static void print_dot_header(struct uftrace_dump_ops *ops,
+			     struct ftrace_file_handle *handle,
+			     struct opts *opts)
+{
+	pr_out("# version\":\"uftrace %s\",\n", UFTRACE_VERSION);
+
+	if (handle->hdr.info_mask & (1UL << CMDLINE))
+		pr_out("# command_line \"%s\"\n", handle->info.cmdline);
+
+	pr_out("digraph \"");
+	print_flame_header(ops, handle, opts);
+	pr_out("%s", (&handle->info)->exename);
+	pr_out("\" { \n");
+	pr_out("\n\t# Attributes \n");
+	pr_out("\tsplines=ortho;\n");
+	pr_out("\tconcentrate=true;\n");
+	pr_out("\tnode [shape=\"rect\",fontsize=\"7\",style=\"filled\"];\n");
+	pr_out("\tedge [fontsize=\"7\"];\n\n");
+}
+
+static void print_dot_task_start(struct uftrace_dump_ops *ops,
+				 struct ftrace_task_handle *task)
+{
+	print_flame_task_start(ops, task);
+}
+
+static void print_dot_inverted_time(struct uftrace_dump_ops *ops,
+				    struct ftrace_task_handle *task)
+{
+}
+
+static void print_dot_task_rstack(struct uftrace_dump_ops *ops,
+				  struct ftrace_task_handle *task, char *name)
+{
+	print_flame_task_rstack(ops, task, name);
+}
+
+static void print_dot_task_event(struct uftrace_dump_ops *ops,
+				 struct ftrace_task_handle *task)
+{
+}
+
+static void print_dot_kernel_start(struct uftrace_dump_ops *ops,
+				   struct uftrace_kernel_reader *kernel)
+{
+}
+
+static void print_dot_cpu_start(struct uftrace_dump_ops *ops,
+				struct uftrace_kernel_reader *kernel, int cpu)
+{
+}
+
+static void print_dot_kernel_rstack(struct uftrace_dump_ops *ops,
+				    struct uftrace_kernel_reader *kernel, int cpu,
+				    struct uftrace_record *frs, char *name)
+{
+}
+
+
+static void print_dot_kernel_event(struct uftrace_dump_ops *ops,
+				   struct uftrace_kernel_reader *kernel, int cpu,
+				   struct uftrace_record *frs)
+{
+}
+
+static void print_dot_kernel_lost(struct uftrace_dump_ops *ops,
+				    uint64_t time, int tid, int losts)
+{
+}
+
+static void print_dot_perf_start(struct uftrace_dump_ops *ops,
+				   struct uftrace_perf_reader *perf, int cpu)
+{
+}
+
+static void print_dot_perf_event(struct uftrace_dump_ops *ops,
+				   struct uftrace_perf_reader *perf,
+				   struct uftrace_record *frs)
+{
+}
+
+static void print_fg_to_dot(struct uftrace_graph_node *node,
+				struct opts *opts)
+{
+	struct uftrace_graph_node *child;
+	unsigned long sample = node->nr_calls;
+
+	if (sample && opts->sample_time)
+		sample = (node->time - node->child_time) / opts->sample_time;
+
+	if (sample) {
+		struct uftrace_graph_node *parent = node->parent;
+
+		pr_out("\t");
+		if (parent != NULL && parent->name != NULL) {
+			pr_out("%s", parent->name);
+			pr_out("->");
+		}
+		pr_out("%s", node->name);
+
+		// Edge Attributes
+		pr_out("[");
+		pr_out("xlabel = \"Calls : %lu\"", sample);
+		pr_out("]");
+		pr_out("\n");
```

Comment:
Hmm.. why not making it a single line like `pr_out("[xlabel = \"Calls: %lu\"]\n", sample);`?